token-metadata/program/ProgrammableNFTGuide.md

```diff
@@ -69,22 +73,115 @@ pub enum MetadataInstruction {
+    * [X] Creation of Fungible Tokens (*fungible tokens*)
+- [ ] `Delegate`
+    * [ ] Creation of `Authority` delegates
+    * [X] Creation of `Collection` delegates
+    * [X] Creation of `Sale` delegates
+    * [X] Creation of `Transfer` delegates
+    * [X] Creation of `Update` delegates
+    * [ ] Creation of `Use` delegates
+    * [X] Creation of `Utility` delegates
+- [ ] `Migrate`
+- [X] `Mint`
+    * [X] Mint Programmable Non-Fungibles tokens (pNFT)
+    * [X] Mint of Non-Fungibles tokens (NFT)
+    * [X] Mint Fungible Assets (*semi-fungible tokens*)
+    * [X] Mint Fungible Tokens (*fungible tokens*)
+- [ ] `Print`
+    * [ ] Print of editions
+- [ ] `Revoke`
+    * [ ] Revoke of `Authority` delegates
+    * [X] Revoke of `Collection` delegates
+    * [X] Revoke of `Sale` delegates
+    * [X] Revoke of `Transfer` delegates
+    * [X] Revoke of `Update` delegates
+    * [ ] Revoke of `Use` delegates
+    * [X] Revoke of `Utility` delegates
+- [X] `Transfer`
+    * [X] wallet-to-wallet transfers
+    * [X] wallet-to-program transfers
+    * [X] program-to-wallet transfers
+- [X] `Update`
+    * [X] Update metadata details for Programmable Non-Fungibles
+    * [X] Update metadata details for Non-Fungibles
+    * [X] Update metadata details for Fungibles Assets
+    * [X] Update metadata details for Fungibles
+- [ ] `Utility`
+    * [X] Lock/Unlock Programmable Non-Fungibles
+    * [X] Lock/Unlock Non-Fungibles
+    * [ ] Authorize use of an asset
+- [ ] `Verify`
+    * [ ] Verify collection items
+    * [ ] Verify creators
+
+## Positional Optional Accounts
+
+The new instruction handlers support positional optional accounts, where an account can be present or not in a transaction. When a instruction is created, it is necessary to provide a list of `PublicKeys` for the instruction accounts – e.g.:
+```javascript
+const mintAcccounts: MintInstructionAccounts = {
+    token,
+    tokenOwner,
+    metadata,
+    masterEdition,
+    mint,
+    payer,
+    ...
+};
+```
+In general, the accounts will be added to the transaction following a pre-defined position:
+```javascript
+// Accounts relative position:
+0. token
+1. tokenOwner
+2. metadata
+3. masterEdition
+4. mint
+5. payer
+...
+```
+When you are minting from a semi-fungible token, there is no need to pass a `masterEdition` account (semi-fungibles do not have a master edition account associated). If we simple ommit the `masterEdition` account, the relative position of the remaining accounts (the accounts after the master edition) would change, resulting in the program logic to be inconsistent. One way to address this is to set another `PublicKey` value to represent a "not-set-value" to maintain the position but at the same time indicate that the master edition account was not set. This is accomplished by setting the Token Metadata program key as the `PublicKey` for any account that should be ommited. This is an efficient approach since:
```

Comment:
/sp "simple ommit" should be "simply omit"

Reply:
Also I suppose you could also mention that this is somewhat of a standard approach?  I believe it is as Solana has suggested it to me on a different program.  Feel free to not say that though, it was just a thought.